_posts/nvola/2017-05-22-turtlehacks.md

```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+author: nvola
+title: "Natasha's Turtlehack"
+---
+Here is my Turtlehack trinket. It requests a user's input for a color and a shape and creates a spiral pattern based on their input.
+
+<iframe src="https://trinket.io/embed/python/b169535972?start=result" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+
+
+I got the idea from this based on the Turtle Basics assignment; I really like the idea of loops and like the way that spirals look, so I thought it would be cool to not only figure out how to code loops to build spirals but to also incorporate some user input so that the user could personalize this a bit. 
+
+
+The first thing I coded was the user input aspect; the Chapter 2 reading and exercises made this part very straightforward and easy to adapt to what I wanted to do. One thing that I found challenging was creating the loops (reasonably so, given that we haven't yet covered that topic). The first reading material I came across after looking online for loops in Python suggested using WHILE statements. These were pretty easy to figure out, and having taken some coding courses in the past I have a vague knowledge of how loops work so it wasn't difficult for me to figure out. What was difficult was dealing with the syntax; the material I found online did not explicitly explain that the WHILE statement required a colon so when I forgot to include the colon it was a little confusing about what was causing the error. I am also still a little unclear about the difference between FOR and WHILE loops, which I'm sure we'll get to soon. What took me the most time in creating these loops was more aesthetic than technical: figuring out how many iterations per size, what the base size should be, what the increase in size each iteration should be, etc.
```

Comment:
Nice discussion of your process here, and how you worked through these concepts.

Spoiler: `for` loops are definite, which means they go a finite amount of times, whereas `while` loops are indefinite, which means they go on until broken or their condition becomes false.  `for` loops are used with lists or generators that produce a list of things.